### PR TITLE
Move Renderer behind an interface

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* This is a non-functional refactor that moves our template renderer behind a nice interface
    * renderer.go is really easy to read now to see what it supports
* Add a test-renderer using [Template Check](https://pkg.go.dev/github.com/jba/templatecheck). This will allow us to create unit-tests that validate golang template rendering during tests (currently only fail at runtime)

### CODE REVIEW

File: pkg/render/renderer_impl.go [hunk 1]
```diff
@@ -71,11 +70,13 @@ type Renderer struct {
 	templatesRoot string
 }
 
+var _ Renderer = (*Impl)(nil) // ensure interface satisfied
+
 // New creates a new renderer with the given details.
-func New(ctx context.Context, root string, debug bool) (*Renderer, error) {
+func New(ctx context.Context, root string, debug bool) (*Impl, error) {
```
Comment (mikehelmick) [inline pkg/render/renderer_impl.go:76]: don't you want the return type to be `Renderer`?

File: pkg/render/renderer_impl.go [hunk 2]
```diff
@@ -49,9 +48,9 @@ var allowedResponseCodes = map[int]struct{}{
 	500: {},
 }
 
-// Renderer is responsible for rendering various content and templates like HTML
-// and JSON responses.
-type Renderer struct {
+// Impl is responsible for rendering various content and templates like HTML
```
Comment (sethvargo) [inline pkg/render/renderer_impl.go:51]: Maybe "Production" could be a better name?

File: pkg/render/renderer_impl.go (hunk 1 above)
Comment (sethvargo) [reply pkg/render/renderer_impl.go:76]: +1

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: pkg/render/renderer_impl.go (hunk 1 above)
Comment (whaught) [reply pkg/render/renderer_impl.go]: Done

File: pkg/controller/codes/bulk_issue_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	"github.com/google/exposure-notifications-verification-server/pkg/config"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/codes"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/render"
+	"github.com/google/exposure-notifications-verification-server/pkg/sms"
+)
+
+func TestShowBulkIssueCodes(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (whaught) [reply pkg/controller/codes/bulk_issue_test.go]: This is intended to test the rendering of the 'show' page template. This allows us to catch issues in the template before runtime (lighter-weight than our full headless chrome tests)

File: pkg/controller/codes/bulk_issue_test.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ func TestShowBulkIssueCodes(t *testing.T) {
 	db.SaveRealm(realm, database.SystemTest)
 
 	config := &config.ServerConfig{}
-	h, err := render.NewTest(ctx, "", t)
+	h, err := render.NewTest(ctx, "../../../cmd/server/assets", t)
```
Comment (sethvargo) [inline pkg/controller/codes/bulk_issue_test.go:43]: Use project.Root instead please

Comment (whaught) [reply pkg/controller/codes/bulk_issue_test.go]: aha that's exactly what I was looking for

**Comment by sethvargo** [comment]
This LGTM, but there's a merge conflict

File: pkg/controller/codes/bulk_issue_test.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/google/exposure-notifications-verification-server/pkg/render"
+	"github.com/google/exposure-notifications-verification-server/pkg/sms"
+)
+
+func TestRenderBulkIssue(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	db, _ := testDatabaseInstance.NewDatabase(t, nil)
+	realm := database.NewRealmWithDefaults("Test Realm")
```
Comment (sethvargo) [inline pkg/controller/codes/bulk_issue_test.go:37]: Should this be a clicky-clicky test? Either way. there's a default realm and user (created by migrations) that you can use to make the test cleaner and more succinct.

File: pkg/render/renderer_impl.go [hunk 6]
```diff
@@ -74,11 +73,17 @@ type Renderer struct {
...
 
+var _ Renderer = (*ProdRenderer)(nil) // ensure interface satisfied
+
 // New creates a new renderer with the given details.
-func New(ctx context.Context, root string, debug bool) (*Renderer, error) {
+func New(ctx context.Context, root string, debug bool) (Renderer, error) {
+	return newImpl(ctx, root, debug)
+}
+
+func newImpl(ctx context.Context, root string, debug bool) (*ProdRenderer, error) {
```
Comment (sethvargo) [inline pkg/render/renderer_impl.go:83]: Nit: `newProdRenderer`

**Review by sethvargo — APPROVED** [review]
/lgtm

/hold

for one nit. Feel free to unhold and do a follow-up, or fix and add the lgtm label manually.

**Pull request merged**